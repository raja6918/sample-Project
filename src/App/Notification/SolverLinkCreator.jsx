import React from 'react';
import { Link } from 'react-router-dom';
import storage from '../../utils/storage';
import PropTypes from 'prop-types';
import { closeNotificationBar } from '../../actions/generic';
import { connect } from 'react-redux';
import { READ_ONLY } from '../../constants';

export const SolverLinkCreator = props => {
  const path = props.href.split('/');
  const pathname = '/solver/' + Number(path[path.length - 2]);
  let readOnly = false;
  const openedScenario = storage.getItem('openScenario');
  if (openedScenario) {
    readOnly = openedScenario.status === READ_ONLY;
  }

  const state = {
    openSolverId: Number(path[path.length - 1]),
    openItemId: Number(path[path.length - 2]),
    openItemName: openedScenario ? openedScenario.name : '',
    readOnly,
  };
  const name = Array.isArray(props.children)
    ? props.children[0].props.value
    : 'solvername';
  const jsx = (
    <Link
      to={{ pathname, state }}
      style={{ textDecoration: 'none', color: '#0a75c2', fontSize: '13px' }}
      onClick={() => props.closeNavNotificationBar(true)}
    >
      {name}
    </Link>
  );
  return jsx;
};

SolverLinkCreator.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.arrayOf(PropTypes.object),
  closeNavNotificationBar: PropTypes.func.isRequired,
};
const mapStateToProps = state => {
  return state;
};
const mapDispatchToProps = dispatch => {
  return {
    closeNavNotificationBar: bool => dispatch(closeNotificationBar(bool)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SolverLinkCreator);
