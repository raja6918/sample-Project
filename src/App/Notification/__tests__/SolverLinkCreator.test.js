import React from 'react';
import { shallow } from 'enzyme';
import { SolverLinkCreator } from '../SolverLinkCreator';

const minProps = {
  href: '/solver/112/3',
  children: [{ props: { value: 'solverName' } }],
};
let wrapper;
describe(`Test suite to test the SolverLinkCreator component`, () => {
  beforeEach(() => {
    wrapper = shallow(<SolverLinkCreator {...minProps} />);
  });
  test('SolverLinkCreator component renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
