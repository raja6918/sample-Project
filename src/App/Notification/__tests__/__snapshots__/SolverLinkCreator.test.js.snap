// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite to test the SolverLinkCreator component SolverLinkCreator component renders 1`] = `
<Link
  onClick={[Function]}
  replace={false}
  style={
    Object {
      "color": "#0a75c2",
      "fontSize": "13px",
      "textDecoration": "none",
    }
  }
  to={
    Object {
      "pathname": "/solver/112",
      "state": Object {
        "openItemId": 112,
        "openItemName": "",
        "openSolverId": 3,
        "readOnly": false,
      },
    }
  }
>
  solverName
</Link>
`;
